src/google/protobuf/compiler/php/php_generator.cc

```diff
@@ -116,31 +116,23 @@ std::string RenameEmpty(const std::string& name) {
   }
 }
 
-std::string MessageFullName(const Descriptor* message, bool is_descriptor) {
-  if (is_descriptor) {
-    return StringReplace(message->full_name(),
-                         "google.protobuf",
-                         "google.protobuf.internal", false);
-  } else {
-    return message->full_name();
-  }
-}
-
-std::string EnumFullName(const EnumDescriptor* envm, bool is_descriptor) {
+template <typename DescriptorType>
+std::string DescriptorFullName(const DescriptorType* desc, bool is_descriptor) {
   if (is_descriptor) {
-    return StringReplace(envm->full_name(),
+    return StringReplace(desc->full_name(),
                          "google.protobuf",
                          "google.protobuf.internal", false);
   } else {
-    return envm->full_name();
+    return desc->full_name();
   }
 }
 
 template <typename DescriptorType>
 std::string ClassNamePrefix(const string& classname,
-                            const DescriptorType* desc) {
+                            const DescriptorType* desc,
+                            bool add_php_prefix) {
```

Comment:
Why do we need add_php_prefix? Don't we have all info in desc?

Reply:
It's not about if the classname needs a prefix, it's about whether to add the prefix even if it's specified. This is because this function is used to escape PHP reserved names, and there are cases in the namespaces (e.g. `PBEmpty\PBEcho\PrefixTestPrefix`) where we want a prefix to prevent a reserved name, but we do not want the PHP prefix added.

This is not actually related to the nested namespace enhancement, and is actually a separate bug fix for escaping namespaces.